crates/core/src/subscription/module_subscription_actor.rs

```diff
@@ -202,29 +197,35 @@ impl ModuleSubscriptionActor {
         })
     }
 
-    async fn broadcast_commit_event(&mut self, event: ModuleEvent) -> Result<(), DBError> {
-        async fn _broadcast_commit_event(
-            auth: AuthCtx,
-            mut event: ModuleEvent,
-            subscription: SubscriptionRw,
-            relational_db: Arc<RelationalDB>,
-        ) -> Result<(), DBError> {
-            let ctx = ExecutionContext::incremental_update(relational_db.address());
-            let mut tx = relational_db.begin_tx();
-            let database_update = event.status.database_update().unwrap();
-            let subscription = subscription.read_arc();
-            let futures = FuturesUnordered::new();
+    fn broadcast_commit_event(&mut self, event: ModuleEvent) -> Result<impl Future<Output = ()> + '_, DBError> {
+        let auth = AuthCtx::new(self.owner_identity, event.caller_identity);
 
-            let incr = subscription
-                .queries
-                .eval_incr(&relational_db, &mut tx, database_update, auth)?;
+        let futures = FuturesUnordered::new();
+
+        let iter = self.subscriptions.par_iter_mut().map_init(
+            || {
+                scopeguard::guard(self.relational_db.begin_tx(), |tx| {
+                    let ctx = ExecutionContext::incremental_update(self.relational_db.address());
+                    self.relational_db.release_tx(&ctx, tx);
+                })
+            },
+            |tx, subscription| -> Result<_, DBError> {
+                let database_update = event.status.database_update().unwrap();
+                let incr = subscription
+                    .queries
+                    .eval_incr(&self.relational_db, tx, database_update, auth)?;
+                Ok((subscription, incr))
+            },
+        );
+        let iter = iter.map(|r| -> Result<(), DBError> {
+            let (subscription, incr) = r?;
```

Comment:
I'm not sure what the proper behavior is here, so I'm being conservative, but maybe someone else knows - if one subscriber's eval_incr fails, should we just log it and carry on? Before, it short-circuited the whole thing so that no one got a subscription update, but that doesn't seem super correct.